solr/core/src/java/org/apache/solr/parser/SolrQueryParserBase.java

```diff
@@ -330,6 +339,19 @@ public void setAllowSubQueryParsing(boolean allowSubQueryParsing) {
     this.allowSubQueryParsing = allowSubQueryParsing;
   }
 
+  /**
+   * Set how overlapping query terms should be scored, as if they're the same term,
```

Comment:
I think some reference to "synonyms" here would be helpful to people understanding, even if this applies to cases that aren't necessarily synonyms in the strict sense.  For example after "overlapping query terms" add a parenthetical: "(e.g. synonyms)"

Heck, maybe we should call this `SynonymQueryStyle`?  After all, we're overriding `newSynonymQuery` to do the work, thus Lucene has picked the name for us.